### PR TITLE
Upgrade modules and support go 1.20

### LINKED ISSUE #774: Support Go 1.20
I tried running receptor on fedora 38 and saw this:

```
shanemcd@fedora38:~/receptor
devel ✗ $ make
CGO_ENABLED=0 GOFLAGS="-buildvcs=false" go build -o receptor  -ldflags "-X 'github.com/ansible/receptor/internal/version.Version=v1.4.0.dev2+g50db04b'"  ./cmd/receptor-cl
# github.com/lucas-clemente/quic-go/internal/qtls
../go/pkg/mod/github.com/lucas-clemente/quic-go@v0.28.1/internal/qtls/go120.go:6:13: cannot use "The version of quic-go you're using can't be built on Go 1.20 yet. For more details, please see https://github.com/lucas-clemente/quic-go/wiki/quic-go-and-Go-versions." (untyped string constant "The version of quic-go you're using can't be built on Go 1.20 yet. F...) as int value in variable declaration
```

### PR DESCRIPTION
**Pull request by ToxicGLaDOS**
Closes #774
In [v0.32.0](https://github.com/quic-go/quic-go/releases/tag/v0.32.0) of quic-go they switched to their own organization instead of lucas-clemente's personal account, hence the change to pkg/netceptor/conn.go.

### CODE REVIEW

**Comment by ToxicGLaDOS** [comment]
quic-go dropped support for Go 1.18 in [v0.33.0](https://github.com/quic-go/quic-go/releases/tag/v0.33.0), so it might make sense to update the automation to check against Go 1.20 instead of Go 1.18 in this pr too.

**Comment by AaronH88** [comment]
Thanks for the PR,

Can you change this file to [1.19, 1.20] please? : https://github.com/ansible/receptor/blob/devel/.github/workflows/pull_request.yml#L42

**Comment by ToxicGLaDOS** [comment]
> Thanks for the PR,
> 
> Can you change this file to [1.19, 1.20] please? : https://github.com/ansible/receptor/blob/devel/.github/workflows/pull_request.yml#L42

Done! But do we also want to update [build_binary_from_ref.yml](https://github.com/ansible/receptor/blob/devel/.github/workflows/build_binary_from_ref.yml#LL21C5-L21C5)
[promote.yml](https://github.com/ansible/receptor/blob/devel/.github/workflows/promote.yml#L85) to 1.20? 1.19 should still work, but it seems we were doing the latest working version previously.

**Comment by AaronH88** [comment]
> > Thanks for the PR,
> > Can you change this file to [1.19, 1.20] please? : https://github.com/ansible/receptor/blob/devel/.github/workflows/pull_request.yml#L42
> 
> Done! But do we also want to update [build_binary_from_ref.yml](https://github.com/ansible/receptor/blob/devel/.github/workflows/build_binary_from_ref.yml#LL21C5-L21C5) [promote.yml](https://github.com/ansible/receptor/blob/devel/.github/workflows/promote.yml#L85) to 1.20? 1.19 should still work, but it seems we were doing the latest working version previously.

I think your correct, I missed that part!
Are you ok to add this?

**Comment by ToxicGLaDOS** [comment]
@AaronH88 Yeah, I can add that. Do you know why the CI is saying "Go 1.2" instead of "Go 1.20"?

**Comment by AaronH88** [comment]
> @AaronH88 Yeah, I can add that. Do you know why the CI is saying "Go 1.2" instead of "Go 1.20"?

Thanks!

No, I am currently searching now to see if the `go-version` field is being treated as a float

File: .github/workflows/pull_request.yml
Comment (AaronH88) [inline .github/workflows/pull_request.yml]: From https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html

YAML converts certain strings into floating-point values, such as the string 1.0. If you need to specify a version number (in a requirements.yml file, for example), you will need to quote the value if it looks like a floating-point value:

version: "1.0"

Can we put 1.19 and 1.20 in quotes please

**Comment by AaronH88** [comment]
Looking great @ToxicGLaDOS , thanks so much!

The linter has shown some issues now that we are using 1.20 . This will explode the size of this PR.

If you dont have time to go through these linting issues, thats all good, I can take over from here later in the week.
Let me know how youd like to proceed, and thanks again! :smile:

**Comment by ToxicGLaDOS** [comment]
@AaronH88 I should be able to fix the linter issues as long as they don't require any structural changes to the code. If I run out of time to work on this I'll post again and let ya know.

**Comment by ToxicGLaDOS** [comment]
@AaronH88 Okay! I pushed some updates, but there were a few changes that I'm unsure about that I'd like someone else who's more familiar with the codebase to look at. Specifically

1. The changes to `_` for unused variables, can those parameters just be removed from the function definition?
2. Is [Context.Background()](https://github.com/ansible/receptor/pull/779/commits/eebb57c87b629832bfcf2dc68f85ad787e6f015b#diff-792611baeb730234abface9c9bc33204ea86453469e7fcdfbe8de8d4e04f2598R455) the correct context?
3. I capitalized [BackendInfo](https://github.com/ansible/receptor/pull/779/commits/eebb57c87b629832bfcf2dc68f85ad787e6f015b#diff-34e0b90451affd6f30e2a36919a995c319236ef547e1d3e44c060d1700bd01deR269) to export it but we could've also not exported the functions that returned it (`TCPListen`, `UDPListen`, and `WebsocketListen`) as far as I can tell we don't use those functions outside of the package, but maybe it's exported for external use 🤷 
4. I only explicitly ignored [one linter issue](https://github.com/ansible/receptor/pull/779/commits/eebb57c87b629832bfcf2dc68f85ad787e6f015b#diff-e014010fd9a9e6fa761fb46bc1ca5f62bea8b3b699cec05c8ae75c3296b47834R186) because the comment made it sound like that's what we wanted. I'd just like to confirm that that's still true.

**Comment by AaronH88** [comment]
> @AaronH88 Okay! I pushed some updates, but there were a few changes that I'm unsure about that I'd like someone else who's more familiar with the codebase to look at. Specifically
> 
> 1. The changes to `_` for unused variables, can those parameters just be removed from the function definition?
> 2. Is [Context.Background()](https://github.com/ansible/receptor/pull/779/commits/eebb57c87b629832bfcf2dc68f85ad787e6f015b#diff-792611baeb730234abface9c9bc33204ea86453469e7fcdfbe8de8d4e04f2598R455) the correct context?
> 3. I capitalized [BackendInfo](https://github.com/ansible/receptor/pull/779/commits/eebb57c87b629832bfcf2dc68f85ad787e6f015b#diff-34e0b90451affd6f30e2a36919a995c319236ef547e1d3e44c060d1700bd01deR269) to export it but we could've also not exported the functions that returned it (`TCPListen`, `UDPListen`, and `WebsocketListen`) as far as I can tell we don't use those functions outside of the package, but maybe it's exported for external use shrug
> 4. I only explicitly ignored [one linter issue](https://github.com/ansible/receptor/pull/779/commits/eebb57c87b629832bfcf2dc68f85ad787e6f015b#diff-e014010fd9a9e6fa761fb46bc1ca5f62bea8b3b699cec05c8ae75c3296b47834R186) because the comment made it sound like that's what we wanted. I'd just like to confirm that that's still true.

Amazing!

Ill reply in line :smile:

File: cmd/receptor-cl/receptor.go
Comment (AaronH88) [inline cmd/receptor-cl/receptor.go:23]: `Start` is an interface so this is correct!

File: pkg/controlsvc/connect.go
Comment (AaronH88) [inline pkg/controlsvc/connect.go:77]: All `ControlFunc` are interfaces so leaving the signature alone is perfect.

This however highlights that we dont check the context for `ctx.Done()`.
I will need to add this to the backlog as it is outside the scope of this PR

File: pkg/controlsvc/reload.go
Comment (AaronH88) [inline pkg/controlsvc/reload.go:140]: both `Init*` 's are interfaces so this is also correct

File: pkg/netceptor/conn.go
Comment (AaronH88) [inline pkg/netceptor/conn.go:493]: this is also an implementation of an interface, the `tls.Config` interface in crypto.go, so again, this is correct

File: pkg/netceptor/external_backend.go
Comment (AaronH88) [inline pkg/netceptor/external_backend.go:95]: Interface, this is correct

File: pkg/netceptor/external_backend.go
Comment (AaronH88) [inline pkg/netceptor/external_backend.go:126]: Interface, this is correct

File: pkg/netceptor/packetconn.go
Comment (AaronH88) [inline pkg/netceptor/packetconn.go:232]: interface

File: pkg/workceptor/kubernetes.go
Comment (AaronH88) [inline pkg/workceptor/kubernetes.go]: Nice, `kw` has the correct context, please use `kw.ctx` :smile:

File: pkg/workceptor/remote_work.go
Comment (AaronH88) [inline pkg/workceptor/remote_work.go]: Nice, this can be removed :smile: 

Also the calls to it can remove the final bool please

File: pkg/workceptor/workunitbase.go
Comment (AaronH88) [inline pkg/workceptor/workunitbase.go:120]: interface, this is correct

File: tests/functional/mesh/lib.go
Comment (AaronH88) [inline tests/functional/mesh/lib.go:756]: This seems fine :smile:

File: tests/functional/mesh/mesh_test.go
Comment (AaronH88) [inline tests/functional/mesh/mesh_test.go:186]: This is still true, nice work

**Comment by AaronH88** [comment]
Great work here! Some small tweaks and we should be ready :smile:

**Comment by ToxicGLaDOS** [comment]
@AaronH88 Okay I think that covers the issues. Let me know if there's more to be done!

**Comment by ToxicGLaDOS** [comment]
Huh... I see a linter error has appeared. I have the linter set to a pre-commit hook and I ran it manually before I committed. Maybe I'm on a different version of the linter? Anyways, I'll fix that up and squash it into the previous commit.

**Comment by ToxicGLaDOS** [comment]
Oh but the CI doesn't give me a line number 🤔

**Comment by AaronH88** [comment]
> Oh but the CI doesn't give me a line number thinking

I think I can see more than you:

Check failure on line 328 in pkg/workceptor/kubernetes.go 
GitHub Actions
/ lint-receptor

return with no blank line before (nlreturn)

**Comment by ToxicGLaDOS** [comment]
> > Oh but the CI doesn't give me a line number thinking
> 
> I think I can see more than you:
> 
> Check failure on line 328 in pkg/workceptor/kubernetes.go GitHub Actions / lint-receptor
> 
> return with no blank line before (nlreturn)

Huh... I'm not sure what to make of that. There isn't a return on that line? That seems to be the end of the `kw.Warning(...)` call.

**Comment by AaronH88** [comment]
> > > Oh but the CI doesn't give me a line number thinking
> > 
> > 
> > I think I can see more than you:
> > Check failure on line 328 in pkg/workceptor/kubernetes.go GitHub Actions / lint-receptor
> > return with no blank line before (nlreturn)
> 
> Huh... I'm not sure what to make of that. There isn't a return on that line? That seems to be the end of the `kw.Warning(...)` call.

That is very strange! 
Can you add a line break to see if it resolves please?

If it dosent, it might be a false positive and we should ignore.

**Comment by ToxicGLaDOS** [comment]
I added a newline before the kw.Warning call. If that doesn't work I'd like to try rebasing onto devel. Maybe it has something to do with an interaction between my commit and something that's on devel that I don't have on my local machine yet 🤷.

**Comment by ToxicGLaDOS** [comment]
Ah it was the rebase after all! I think this one will work 😅

**Comment by AaronH88** [comment]
> Ah it was the rebase after all! I think this one will work sweat_smile

Excellent! 

Ill give it one final review in the morning and merge, thanks so much for this!

**Review by AaronH88 — APPROVED** [review]
lgtm

**Comment by AaronH88** [comment]
Sorry for the delay in this, I needed to get more permissions to change the required tests for our workflows.

I also spoke with our Ansible stakeholders to ensure that this wont cause any problems and I have been asked to make a release now, before this PR gets merged.
So, I should be in a position to merge this in the next few days once this is all done :smile:

**Pull request merged**
